### ISSUE TITLE
Upload error on Staging and Production.

### ISSUE THREAD

**Issue by mattbaumann1** [NUARIG/competitions#849]
Cannot upload documents for a submission.

UI response on upload.
"Error storing "dummy1.pdf". Status: 0"

logs:
INFO -- : Disk Storage (0.6ms) Uploaded file to key: rbdcx80uudgirghykmgmjv2fdgpv (checksum: )
INFO -- : No template found for ActiveStorage::DiskController#update, rendering head :no_content
INFO -- : Completed 204 No Content in 2ms (ActiveRecord: 0.0ms | Allocations: 661)
INFO -- : Started GET "/profile/reviews" for 99.138.104.25 at 2021-07-30 10:11:51 -0500

**Comment by mattbaumann1** [comment]
We determined that this is only an issue with safari. 

@fanglinnw suggested the following fix.
```
class UploadController < ActiveStorage::DiskController
  after_action :set_content_length_header, only: :update
  # This is a workaround due to a Passenger bug in 5.0.20. Passenger fixed this in 5.0.22.
  # https://github.com/phusion/passenger/issues/1595
  # The bug is a that Passenger automatically add a `Transfer-Encoding: chunked` header when sending 204 No Content responses,
  # but this is not allowed according to the HTTP 1.1 spec.
  # Some clients such as Safari or Curl will time out and fail when parsing the 204 response with this header.
  # The workaround here is to return 200 OK instead.
  def set_content_length_header
    if response.status == 204
      response.status = 200
    end
  end
end
```

**Comment by mattbaumann1** [comment]
Currently exploring what to do about this. It looks like it might make sense to have FSMIT upgrade passenger.

**Comment by mattbaumann1** [comment]
We are having some issues with file uploads and ActiveStorage. It looks like it is an issue in Passenger 5.0.20 and fixed in 5.0.22. 
Here are two relevant tickets.
 
https://github.com/phusion/passenger/issues/1595
https://github.com/phusion/passenger/issues/1517

The current version of Passenger is 6.0.10 (https://github.com/phusion/passenger). When I did a search for the version of Passenger on the server, it is the 5.0.20.

Also find below the pertinent sections of our application logs.
 
INFO -- : Disk Storage (0.6ms) Uploaded file to key: xxxxx (checksum: xxxx)
INFO -- : No template found for ActiveStorage::DiskController#update, rendering head :no_content
INFO -- : Completed 204 No Content in 2ms (ActiveRecord: 0.0ms | Allocations: 661)
INFO -- : Started GET "/profile/reviews" for 99.138.104.25 at 2021-07-30 10:11:51 -0500
